### DISCUSSION TITLE
Default value generation

### DISCUSSION THREAD

**Discussion by dieseldjango** [rjsf-team/react-jsonschema-form#2750] [Q&A]
I've been working some more with the form and it appears that it automatically generates default values for properties missing in the input data if provided by the schema. This seems to make sense and to be useful when the form is used for editing. However, my initial use case for this form is to present a read-only view of the data. When the data is missing a value, the default is presented, which is misleading.

Is there a way to turn off the default value generation?

**Comment by epicfaace** [comment]
I don't think so, at the moment.

--
Ashwin Ramaswami

**Comment by epicfaace** [comment]
You could modify the schema before passing it into the form by removing all
`default` keys?

--
Ashwin Ramaswami

**Comment by dieseldjango** [comment]
Thanks for the answer. I'll look at stripping defaults for now and maybe submit a PR later to support disabling the feature if I find time.

**Comment by epicfaace** [comment]
Thanks!

--
Ashwin Ramaswami
